### PR TITLE
fix post story text backgroundArgb missing after generatedLinkPreview

### PR DESCRIPTION
**Pull request by hasanbasri1993**
before when sendMessage into status@broadcast backgroundArgb will replace after generateLinkPreview

```

let content
        content = {
            text: "Assalamualaikum, selamat ananda Salsabila LULUS tes PSB DaarulUluumLido. | Selanjutnya silahkan bayar pendaftaran dengan cara berikut:\n hanya berlaku 1x24 jam. Datang ke Alfamart ; ingat, hanya Alfamart, bukan yang lain, Sampaikan ke kasir ingin melakukan pembayaran Plasamall, Berikan kode bayar 1001342127666 ke kasir, Bayar sesuai jumlah yang diinfokan oleh kasir, (350ribu + biaya admin yang muncul di Alfamart), https://tripay.co.id/checkout/T1941708004**** Simpan struk bukti pembayaran Anda.\n Jika sudah mendapatkan username, silahkan hubungi Pak Deni Herdayadi di https://wasap.at/Jgo***",
            font: 0,
            textArgb: 4279702806,
            backgroundArgb: 4284659069
        }
        conn.sendMessage("status@broadcast",
            content,
            MessageType.extendedText, {}).then(function (processData) {
            console.log(`🍳 Success ${JSON.stringify(processData)}`)
        });
```

image

### CODE REVIEW

File: src/WAConnection/6.MessagesSend.ts [hunk 1]
```diff
@@ -109,18 +109,25 @@ export class WAConnection extends Base {
     /** Prepares the message content */
     async prepareMessageContent (message: string | WATextMessage | WALocationMessage | WAContactMessage | WAContactsArrayMessage | WAGroupInviteMessage | WAMediaUpload | WAListMessage | WAButtonsMessage, type: MessageType, options: MessageOptions) {
         let m: WAMessageContent = {}
+        let message2
```
Comment (PurpShell) [inline src/WAConnection/6.MessagesSend.ts:112]: Would you mind not adding a new variable? i think we can already use `message` for this

File: src/WAConnection/6.MessagesSend.ts [hunk 2]
```diff
@@ -109,18 +109,25 @@ export class WAConnection extends Base {
...
             case MessageType.text:
             case MessageType.extendedText:
                 if (typeof message === 'string') message = {text: message} as WATextMessage
-                
+                if ('backgroundArgb' in message) message2 = message
                 if ('text' in message) {
                     if (options.detectLinks !== false && message.text.match(URL_REGEX)) {
                         try {
-                            message = await this.generateLinkPreview (message.text)
+                            console.log(message)
```
Comment (PurpShell) [inline src/WAConnection/6.MessagesSend.ts:121]: remove logger

**Review by PurpShell — COMMENTED** [review]
some minor changes, also what if you fix the issue (s) with generateLinkPreview instead of re-setting the properties, because maybe greater information is being lost

File: src/WAConnection/6.MessagesSend.ts (hunk 1 above)
Comment (hasanbasri1993) [reply src/WAConnection/6.MessagesSend.ts:112]: How about this ?


```
   /** Query a string to check if it has a url, if it does, return required extended text message */
    async generateLinkPreview(message: proto.ExtendedTextMessage | proto.ButtonsMessage) {
        
        const query = ['query', {type: 'url', url: message.text, epoch: this.msgCount.toString()}, null]
        const response = await this.query ({json: query, binaryTags: [26, WAFlag.ignore], expect200: true, requiresPhoneConnection: false})

        if (response[1]) response[1].jpegThumbnail = response[2]
        const data = response[1] as WAUrlInfo
        if (!(message instanceof ButtonsMessage)) {
            message.canonicalUrl = data['canonical-url']
            message.matchedText = data['matched-text']
            message.jpegThumbnail = data.jpegThumbnail
            message.description = data.description
            message.title = data.title
            message.previewType = 0
        }
        return message
    }
```

Comment (hasanbasri1993) [reply src/WAConnection/6.MessagesSend.ts:112]: how ?

**Comment by hasanbasri1993** [comment]
please merge @adiwajshing this pull

**Comment by PurpShell** [comment]
you haven't committed your changes...

> please merge @adiwajshing this pull

**Pull request closed without merging**